### PR TITLE
Add method to list repository topics

### PR DESCRIPTION
**Pull request by martinvanzijl**
For #546.

Adds method `GHRepository.listTopics`.

NOTE: The method to update repository topics does not work yet, so I have commented it out. It may not have been implemented in the underlying API. I recommend keeping the issue open until this is working.

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 1]
```diff
@@ -1744,4 +1744,29 @@ String getApiTailUrl(String tail) {
     public GHIssueEvent getIssueEvent(long id) throws IOException {
         return root.retrieve().to(getApiTailUrl("issues/events/" + id), GHIssueEvent.class).wrapUp(root);
     }
+
+    // Only used within getTopics().
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepository.java]: Is this used?

Comment (martinvanzijl) [reply src/main/java/org/kohsuke/github/GHRepository.java]: Sorry, I mean "listTopics()". I've updated the comment now.

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 2]
```diff
@@ -1744,4 +1744,29 @@ String getApiTailUrl(String tail) {
...
+    public List<String> listTopics() throws IOException {
+        Topics topics = root.retrieve().withPreview(MERCY).to(getApiTailUrl("topics"), Topics.class);
+        return topics.names;
+    }
+
+    /**
+     * Set the topics for this repository.
+     * See https://developer.github.com/v3/repos/#replace-all-topics-for-a-repository
+     */
+    // This currently returns a "404" error (as of 30 Oct 2019).
```
[the remark this answers is not in the document]
Comment (martinvanzijl) [reply src/main/java/org/kohsuke/github/GHRepository.java]: I guess it is a bug on their side. I also tried using `curl` which gave a "404" response, too. Shall I raise a query with GitHub support?

I suggest merging the getter method but keeping the issue open, so we know to re-visit later and add the setter method.

[the remark this answers is not in the document]
Comment (bitwiseman) [reply src/main/java/org/kohsuke/github/GHRepository.java]: Sure merging this makes sense.

**Comment by martinvanzijl** [comment]
Thanks!

I also raised a query to GitHub support to ask whether the setter feature is enabled yet.

**Pull request merged**